apps/content-gateway-api/src/app/maintenance/MaintenanceJob.ts

```diff
@@ -0,0 +1,43 @@
+import { ProgramError } from "@banklessdao/util-data";
+import { isLeft } from "fp-ts/lib/Either";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import {
+    AsyncTask,
+    SimpleIntervalJob,
+    SimpleIntervalSchedule,
+} from "toad-scheduler";
+import { Logger } from "tslog";
+import { MaintenanceJobError } from "./errors";
+
+/**
+ * A job designed to perform maintenance tasks on the gateway
+ */
+export interface MaintenanceJob {
+    id: string;
+    run: () => TaskEither<MaintenanceJobError, void>;
+    schedule: SimpleIntervalSchedule;
+}
+
+export type addMaintenanceJob = (
+    job: MaintenanceJob
+) => TaskEither<ProgramError, void>;
+
+export const makeToToadJob =
+    (logger: Logger) =>
+    (job: MaintenanceJob): SimpleIntervalJob => {
+        const task = new AsyncTask(job.id, () => {
+            return job
+                .run()()
+                .then(
+                    (v) => {
+                        if (isLeft(v))
+                            throw v.left.error || new Error("Unknown Cause");
```

Comment:
Will this `throw` bubble out of the context of the `SimpleIntervalJob`?

Reply:
I was able to confirm line 37 catches it